mgr/manager.go

```diff
@@ -31,6 +31,59 @@ type CertServiceManager struct {
 	serviceManager svcmgr.Manager
 }
 
+// Process a spec updating content on disk, taking action as needed.
+// Returns (TTL for PKI, error).  If an error occurs, the ttl is at best
+// a hint to the invoker as to when the next refresh is required- that said
+// the invoker should back off and try a refresh.
+func (csm *CertServiceManager) EnforcePKI(enable_actions bool) (time.Duration, error) {
+	err := csm.CheckDiskPKI()
+	if err != nil {
+		log.Debugf("manager: spec %s, checkdiskpki: %s.  Forcing refresh.", csm, err.Error())
+		csm.ResetLifespan()
+	}
+
+	if err = csm.CheckCA(); err != nil {
+		log.Errorf("manager: the CA for %s has changed, but the service couldn't be notified of the change", csm.Spec.Path)
+	}
+
+	lifespan := time.Duration(0)
+	if !csm.Ready() {
+		log.Infof("manager: queueing %s because it isn't ready", csm.Spec.Path)
```

Comment:
Is this log statement accurate? Is there any queueing going on here?

Reply:
So uh... yeah.  The messages all suck.  I was planning on doing a second pass to clean up the messaging, but I wanted to make some changes to CertServiceManager first (for example, creating a useful String() method).

The messages still roughly align in accuracy, but they all need to be rewritten to drop things like the 'manager: ' prefix.

Your call if you'd prefer I clean it now or get it in a follow up; I'm going to do it either way, just a question of rebase issues for your work since I'm mangling mgr/manager.go a fair bit.

Reply:
Makes sense to me!

Reply:
I've resolved this and the other log critique in the latest revision.